File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> out;
        int m = target.size();
        int j = 0;
        int i = 1;
        while(i <= n && j < m){
            if (target[j] == i){
                j++;
                i++;
                out.push_back("Push");
            }
            else{
                out.push_back("Push");
                out.push_back("Pop");
                i++;
            }
        }
        return out;
    }
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:

        i = 1
        j = 0
        out = []
        while (i <= n and j < len(target)):
            if i == target[j]:
                j += 1
                i += 1
                out.append("Push")
            else:
                i += 1
                out.append("Push")
                out.append("Pop")
        return out